org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/OpenIssueInEclipseJob.java

```diff
@@ -81,7 +90,43 @@ public OpenIssueInEclipseJob(OpenIssueContext context) {
 
   @Override
   protected IStatus run(IProgressMonitor monitor) {
-    // 1) Check if matching between remote and local path does work / branches match
+    // 1) We have to await the analysis getting ready for this project. When also setting up the connection / binding
+    // with this Open in IDE request it isn't the case. We want to display a nice dialog informing the user that they
+    // have to wait for a few more moments.
+    if (!AnalysisReadyStatusCache.getAnalysisReadiness(ConfigScopeSynchronizer.getConfigScopeId(project))) {
+      // Show the console so the user will see the progress!
+      SonarLintUiPlugin.getDefault().getSonarConsole().bringConsoleToFront();
+
+      // We have to run the dialog in the UI thread, therefore we have to work with references here and synchronized
+      // threads for the job to proceed correctly later.
+      var statusRef = new AtomicReference<IStatus>();
+      var cancelledByJob = new AtomicBoolean();
+      statusRef.set(Status.OK_STATUS);
+      cancelledByJob.set(false);
+      Display.getDefault().syncExec(() -> {
+        var dialog = new AwaitProjectConnectionReadyDialog(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell());
+        new AwaitProjectConnectionReadyJob(project, dialog, 0)
+          .schedule(AwaitProjectConnectionReadyJob.SCHEDULE_TIMER_MS);
+        dialog.open();
+        if (dialog.cancelledByJob() || dialog.cancelledByUser()) {
```

Comment:
Can you maybe log if the job got canceled by the user or not? The next message is ambiguous, whether it was a timeout or canceled.

Reply:
Yes, did a debug log